mainwindow.cpp

```diff
@@ -270,6 +276,86 @@ namespace dealii
 
 
 
+    void MainWindow::show_settings ()
+    {
+      settings_dialog = new SettingsDialog(gui_settings,this);
+
+      connect(settings_dialog, SIGNAL(accepted()), this, SLOT(apply_settings()));
+      settings_dialog->exec();
+      disconnect(settings_dialog, SIGNAL(accepted()), this, SLOT(apply_settings()));
+    }
+
+
+
+    void MainWindow::apply_settings ()
+    {
+      update_visible_items();
+      update_font();
+    }
+
+
+
+    void MainWindow::toggle_visible_default_items()
+    {
+      const bool hide_default_values = gui_settings->value("Settings/hideDefault", false).toBool();
+      gui_settings->setValue("Settings/hideDefault", !hide_default_values);
+      update_visible_items();
+    }
+
+
+
+    void MainWindow::update_visible_items()
+    {
+      const bool hide_default_values = gui_settings->value("Settings/hideDefault", false).toBool();
+
+      if (hide_default_values)
+        {
+        for (int i = 0; i < tree_widget->topLevelItemCount(); ++i)
```

Comment:
If not, then ignore the indentation here and use a follow-up patch to just indent the whole thing.